### DISCUSSION TITLE
Zero runtime CSS solution

### DISCUSSION THREAD

**Discussion by pstachula-dev** [mantinedev/mantine#504] [Q&A] [answered]
Is there a possibility in the future to consider Zero-runtime solutions like Linaria or Vanilla-extract?

Is it possible to combine this type of solution with Mantine?

**Comment by rtivital** [comment, marked as answer]
Mantine is designed to be dynamic, this allows us to support many dynamic props like `styles`, `sx`, `mx`, etc. I'm happy with current styles performance and not planning to rewrite the whole library to gain an unnoticeable performance boost. Zero-runtime does not provide any real benifits, see performance comparison between styled components (built with the same styles processor as emotion – stylis) and linaria – https://pustelto.com/blog/css-vs-css-in-js-perf/ – charts are almost identical.

In your applications, you can use any styling solution, Mantine components support styles overrides with anything via `className` and `classNames` props.
